src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/PhysicalPlanFactory.java

```diff
@@ -485,52 +556,101 @@ else if ( lop instanceof LogicalOpBGPAdd ) {
 	public static List<PhysicalPlan> enumeratePlansWithUnaryOpFromReq( final PhysicalOpRequest<?, ?> req,
```

Comment:
Line 551: the operator LogicalOpGPAdd needs to be considered in this function 
```
public static PhysicalPlan extractRequestAsPlan(final UnaryLogicalOp lop) {
...
else if ( lop instanceof LogicalOpGPAdd ) {
	return extractRequestAsPlan( (LogicalOpGPAdd) lop );
}
...
```

Then, the implementation of 'initiateCostEstimation' in CFRNumberOfTermsShippedInRequests can be simplified.